api/v1alpha3/etcdadmconfig_types.go

```diff
@@ -44,6 +44,30 @@ type EtcdadmConfigSpec struct {
 	// +optional
 	EtcdadmBuiltin bool `json:"etcdadmBuiltin,omitempty"`
 
+	// +optional
+	// EtcdadmArgs map[string]interface{} `json:"etcdadmArgs,omitempty"`
+
+	// Format specifies the output format of the bootstrap data
+	// +optional
+	Format Format `json:"format,omitempty"`
+
+	// BottlerocketConfig specifies the configuration for the bottlerocket bootstrap data
+	// +optional
+	BottlerocketConfig *BottlerocketConfig `json:"bottlerocketConfig,omitempty"`
+
+	// CloudConfigConfig specifies the configuration for the cloud-init bootstrap data
+	// +optional
+	CloudConfigConfig *CloudConfigConfig `json:"cloudConfigConfig,omitempty"`
+}
+
+type BottlerocketConfig struct {
+	// EtcdImage specifies the etcd image to use by etcdadm
+	EtcdImage string `json:"etcdImage,omitempty"`
+
+	// BootstrapImage specifies the container image to use for bottlerocket's bootstrapping
+	BootstrapImage string `json:"bootstrapImage"`
+}
+type CloudConfigConfig struct {
```

Comment:
For CAPV we run commands to set hostname through PreEtcdadmCommands, this is done for CAPV through the KCP spec too. Do we not need those commands for BR bootstrap? 
Even if we don't, I think we should keep Pre and PostEtcdadmCommands as top level fields along with Users on the EtcdadmConfigSpec

Reply:
Done